src/org/exist/indexing/IndexController.java

```diff
@@ -331,6 +331,18 @@ public void indexNode(final Txn transaction, final IStoredNode node, final NodeP
     }
 
     /**
+     * Helper method: indexing is starting for a document
+     *
+     * @param transaction the current transaction
+     * @param listener the StreamListener which receives the index events
+     */
+    public void startIndexDocument(final Txn transaction, final StreamListener listener) {
```

Comment:
@adamretter I'm totally lost because IndexController hold StreamListener as class field and pass it as last parameter https://github.com/adamretter/exist/blob/0478bc8240a2d26df8a7360cb261f0d6c03ea9b2/src/org/exist/indexing/IndexController.java#L58